scipy/sparse/linalg/tests/test_interface.py

```diff
@@ -284,6 +284,20 @@ def always_four_ones(x):
         assert_(hasattr(op, "shape"))
         assert_(hasattr(op, "_matvec"))
 
+def matvec(x):
+    """ Needed for test_pickle as local functions are not pickleable """
+    return np.zeros(3)
+
+def test_pickle():
+    import pickle
+
+    N = M = 3
+    A = interface.LinearOperator((M, N), matvec)
+    s = pickle.dumps(A)
+    B = pickle.loads(s)
```

Comment:
As long as the original issue is protocol-dependent, maybe test protocols from `pickle.HIGHEST_PROTOCOL` down to 2